lib/flight_scheduler/submission/job.rb

```
@@ -33,10 +33,13 @@ def initialize(allocation)
     end
 
     def call
-      @job.state = 'RUNNING'
       @allocation.nodes.each do |node|
         initialize_job_on(node)
       end
+      # XXX Should a job transition to RUNNING after all the daemons have been
+      #     notified? Currently the daemons do not report back "job received",
+      #     so doing so reliable is not possible
+      @job.state = 'RUNNING'
```

If the message to the daemon cannot be sent, the `initialize_job_on` method will raise an exception.  So if we get to the `@job.state = 'RUNNING'` line we can be confident that all daemon's have received the message.

Is there something else that we need to do to consider this reliable?

Obviously, there is a pre-existing issue of what to do should one of the allocated nodes be down.  But that is a separate issue than is being worked on here.

---

What are the implications of setting the job's state before starting the batch script?  I'm thinking in particular here if the daemons can be allocated the job, but the batch script cannot be started FSR.  What is the correct state of the job?

---

Right we can probably rely on the underlining TCP socket to ensure the message is delivered. So we can reasonable assume it was submitted correctly if everything exits correctly.

I have moved the state change to the end: 323553bf8545910e213a1d993f2ec0991f88e640

In regards to the `batch_script` failing, it should raise an error a be caught in the existing handler. Not sure if this is correct, so I have added a note about it.